mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -255,6 +255,20 @@ private[spark] object BLAS extends Serializable with Logging {
     }    
   }
 
+  // Workaround for SparseVectors.
+  def syr(alpha: Double, x: SparseVector, A: DenseMatrix) {
+    val mA = A.numRows
+    val nA = A.numCols
+    require(mA == nA, s"A is not a symmetric matrix. A: $mA x $nA")
```

Comment:
Just a general comment -- I find this message a little misleading, since it really only checks for squareness; I realize this is copied from the dense syr() [I found it misleading there, too].